tests/backend/mongodb-ssl/test_ssl_connection.py

```diff
@@ -0,0 +1,306 @@
+        '--sslCAFile=' + certs_dir + '/ca.crt',
+        '--sslCRLFile=' + certs_dir + '/crl.pem',
+        '--sslPEMKeyFile=' + certs_dir + '/test_mdb_ssl_cert_and_key.pem',
+        '--sslPEMKeyPassword=""'
+        ],
+        'ok': 1.0,
+        'parsed': {'replication': {'replSetName': 'bigchain-rs'},
+                   'storage': {'dbPath': '/data'}}
+        }
+
+
+@pytest.fixture
+def mongodb_ssl_connection(certs_dir):
+    import bigchaindb
+    return MongoClient(host=bigchaindb.config['database']['host'],
+                       port=bigchaindb.config['database']['port'],
+                       serverselectiontimeoutms=bigchaindb.config['database']['connection_timeout'],
+                       ssl=bigchaindb.config['database']['ssl'],
+                       ssl_ca_certs=bigchaindb.config['database']['ca_cert'],
+                       ssl_certfile=bigchaindb.config['database']['certfile'],
+                       ssl_keyfile=bigchaindb.config['database']['keyfile'],
+                       ssl_pem_passphrase=bigchaindb.config['database']['keyfile_passphrase'],
+                       ssl_crlfile=bigchaindb.config['database']['crlfile'],
+                       ssl_cert_reqs=CERT_REQUIRED)
+
+
+def test_ssl_get_connection_returns_the_correct_instance(db_host, db_port, certs_dir):
+    from bigchaindb.backend import connect
+    from bigchaindb.backend.connection import Connection
+    from bigchaindb.backend.mongodb.connection import MongoDBConnection
+
+    config = {
+        'backend': 'mongodb',
+        'host': db_host,
+        'port': db_port,
+        'name': 'test',
+        'replicaset': 'bigchain-rs',
+        'ssl': True,
+        'ca_cert':   certs_dir + '/ca.crt',
+        'crlfile':   certs_dir + '/crl.pem',
+        'certfile':  certs_dir + '/test_bdb_ssl.crt',
+        'keyfile':   certs_dir + '/test_bdb_ssl.key',
+        'keyfile_passphrase': ''
+    }
+
+    conn = connect(**config)
+    assert isinstance(conn, Connection)
+    assert isinstance(conn, MongoDBConnection)
+    assert conn.conn._topology_settings.replica_set_name == config['replicaset']
+
+
+@mock.patch('bigchaindb.backend.mongodb.connection.initialize_replica_set')
+@mock.patch('pymongo.MongoClient.__init__')
+@mock.patch('time.sleep')
+def test_ssl_connection_error(mock_sleep, mock_client, mock_init_repl_set):
+    from bigchaindb.backend import connect
+    from bigchaindb.backend.exceptions import ConnectionError
+
+    # force the driver to throw ConnectionFailure
+    # the mock on time.sleep is to prevent the actual sleep when running
+    # the tests
+    mock_client.side_effect = pymongo.errors.ConnectionFailure()
+
+    with pytest.raises(ConnectionError):
+        conn = connect()
+        conn.db
+
+    assert mock_client.call_count == 3
+
+
+@mock.patch('bigchaindb.backend.mongodb.connection.initialize_replica_set')
+@mock.patch('pymongo.MongoClient')
+def test_ssl_connection_run_errors(mock_client, mock_init_repl_set):
```

Comment:
Sure. I just don't think that testing again things like the driver tries to connect 3 times before failing is really relevant since it has nothing to do with ssl